### PR TITLE
CLN: more assorted cleanups

### PR DESCRIPTION
**Pull request by jbrockmendel**
Continuing with the theme of avoiding runtime imports

### CODE REVIEW

File: pandas/core/groupby/generic.py [hunk 1]
```diff
@@ -45,14 +45,16 @@
...
-from pandas.core.index import Index, MultiIndex
+from pandas.core.index import Index, MultiIndex, _all_indexes_same
 import pandas.core.indexes.base as ibase
 from pandas.core.internals import BlockManager, make_block
 from pandas.core.series import Series
 from pandas.core.sparse.frame import SparseDataFrame
 
 from pandas.plotting import boxplot_frame_groupby
 
+from .groupby import groupby
```
Comment (jreback) [inline pandas/core/groupby/generic.py]: can u do absolute imports

File: pandas/core/generic.py [hunk 2]
```diff
@@ -7511,9 +7511,9 @@ def clip(self, lower=None, upper=None, axis=None, inplace=False, *args, **kwargs
         # so ignore
         # GH 19992
         # numpy doesn't drop a list-like bound containing NaN
-        if not is_list_like(lower) and np.any(pd.isnull(lower)):
+        if not is_list_like(lower) and np.any(pd.isna(lower)):
```
Comment (jreback) [inline pandas/core/generic.py]: these are already imported at the top, can drop the pd.

File: pandas/core/groupby/generic.py [hunk 3]
```diff
@@ -35,7 +35,7 @@
     is_object_dtype,
     is_scalar,
 )
-from pandas.core.dtypes.missing import isna, notna
+from pandas.core.dtypes.missing import _isna_ndarraylike, isna, notna
```
Comment (jreback) [inline pandas/core/groupby/generic.py]: future, can you de-privatize this

**Review by jreback — CHANGES_REQUESTED** [review]

File: pandas/core/groupby/generic.py (hunk 1 above)
Comment (jbrockmendel) [reply pandas/core/groupby/generic.py]: yep

**Comment by jbrockmendel** [comment]
Comments addressed

**Review by jreback — APPROVED** [review]

**Comment by jreback** [comment]
thanks @jbrockmendel

**Pull request merged**